engine/access/rpc/backend/backend.go

```diff
@@ -344,61 +348,54 @@ func (b *Backend) GetNodeVersionInfo(_ context.Context) (*accessmodel.NodeVersio
 	return nodeInfo, nil
 }
 
-func (b *Backend) GetCollectionByID(_ context.Context, colID flow.Identifier) (*flow.LightCollection, error) {
-	// retrieve the collection from the collection storage
+// GetCollectionByID returns a light collection by its ID.
+//
+// Expected errors:
+// - access.DataNotFound if the collection is not found.
+func (b *Backend) GetCollectionByID(ctx context.Context, colID flow.Identifier) (*flow.LightCollection, error) {
 	col, err := b.collections.LightByID(colID)
 	if err != nil {
-		// Collections are retrieved asynchronously as we finalize blocks, so
-		// it is possible for a client to request a finalized block from us
-		// containing some collection, then get a not found error when requesting
-		// that collection. These clients should retry.
-		err = rpc.ConvertStorageError(fmt.Errorf("please retry for collection in finalized block: %w", err))
```

Comment:
`backend` shoudl stop returning grpc status codes, and instead return access sentinel errors